cpp/src/toppra/geometric_path.cpp

```diff
@@ -15,8 +15,27 @@ Vectors GeometricPath::eval(const Vector &positions, int order) const {
   };
 
 
-Vector GeometricPath::proposeGridpoints(double max_err_threshold, int max_iteration, double max_seg_length, int min_nb_points) const {
-  std::forward_list<value_type> gridpoints {pathInterval()[0], pathInterval()[1]};
+Vector GeometricPath::proposeGridpoints(double max_err_threshold, int max_iteration, double max_seg_length, int min_nb_points, Vector initialGridpoints) const {
+  std::forward_list<value_type> gridpoints;
+  //gridpoints.reserve(std::max(initialGridpoints.size(), min_nb_points));
+  const Bound I = pathInterval();
+  if (initialGridpoints.size() == 0) {
+    gridpoints.push_front(I[1]);
+    gridpoints.push_front(I[0]);
+  } else {
+    if (initialGridpoints.size() == 1)
+      throw std::invalid_argument("initialGridpoints should be empty or have at least 2 elements");
+    int N = initialGridpoints.size() - 1;
+    for (int i : {0, 1}) {
+      if (std::abs(I[i] - initialGridpoints[i*N]) > TOPPRA_NEARLY_ZERO) {
+        std::ostringstream oss;
+        oss << "initialGridpoints[" << i*N << "] must be " << I[i] << " and not " << initialGridpoints[i*N];
+        throw std::invalid_argument(oss.str());
+      }
+    }
```

Comment:
In addition to this check, we should also check that the given elements are monotonically increasing.